### ISSUE TITLE
Desktop Access - Additional steps required to connect to WS2012R2 as a local user 

### ISSUE THREAD

**Issue by philip-teleport** [gravitational/teleport#26827]
Expected behavior:

Document or automate the additional steps required to support connections to WS2012R2 as a local user using Desktop Access.

Current behavior:

Installing the Teleport service for Windows on WS2012R2 does not fully configure the machine for local user access.

Bug details:
- Teleport version: 12.3.x
- Recreation steps: Install the Teleport service for Windows on WS2012R2.
- Debug logs: N/A

**Comment by zmb3** [comment]
I looked in to simplifying this, as the current instructions we have require openssl and manually copying over a cert to Windows. Here's what I learned:

## Generating Cert

The following code will generate a self-signed certificate and automatically import it into the machine's personal store.

```powershell
$params = @{
    Subject = "CN=$env:computername"
    KeyAlgorithm = 'ECDH_secP384r1'
    KeyExportPolicy = 'Exportable'
    CertStoreLocation = 'Cert:\LocalMachine\My'
    NotAfter = (Get-Date).AddMonths(18)
    Type = 'SSLServerAuthentication'
}

$cert = New-SelfSignedCertificate @params
```

I also looked in to automatically populating the IP SANs with a combination of `Get-NetIpAddress` and the following:

```powershell
   TextExtension = @(
      "2.5.29.17={text}DNS=$env:computername&IPAddress=..."
    )
```

This became a dead-end, as the `KeyAlgorithm` param is not available on 2012R2.

## Permissions

Once you've generated a certificate, you need to give NETWORK SERVICE read access to the cert's private key. I was looking at adapting https://gist.github.com/pharring/991ad7ace955ff455796a6c1009cbca5 for this task. (It's not quite the same, as we're using ECDH keys, not RSA).

## Configuring RDP to use said cert

`Set-RDCertificate` can do this, but Windows doesn't support passing the thumbprint of an existing certificate via `-Thumbprint` until Windows Server 2019.

It looks like we would have to [use `wmic` or update the registry directly](https://learn.microsoft.com/en-us/troubleshoot/windows-server/remote/remote-desktop-listener-certificate-configurations).

## tl;dr

It looks like the options necessary to make a simple script that users could copy paste in order to configure this are not available, and it would probably be easier to build this into the installer.

**Comment by pschisa** [comment]
+1 customer hit this issue with AD joined Windows 2012 R2 server.  RDP connections failed and logging showed a "connection reset by peer" error in the windows desktop service logging right before the failure.

**Comment by zmb3** [comment]
Note: Windows Server 2012R2 will be end of life in less than a month, so it's unlikely that we will pull the team off their current work to continue to support an OS that's no longer supported.

**Comment by zmb3** [comment]
Closing due to 2012R2 being EOL.
